recipe/bld.bat

```diff
@@ -1,93 +1,76 @@
-@echo off
+@echo ON
 
-mkdir build
-cd build
-
-set CMAKE_CONFIG="Release"
-
-:: Set the right msvc version according to Python version
-REM write a temporary batch file to map cl.exe version to visual studio version
-echo @echo 15=9 > msvc_versions.bat
-echo @echo 16=10 >> msvc_versions.bat
-echo @echo 17=11 >> msvc_versions.bat
-echo @echo 18=12 >> msvc_versions.bat
-echo @echo 19=14 >> msvc_versions.bat
+git clone https://github.com/Itseez/opencv_contrib --single-branch --branch %PKG_VERSION% --depth 1
 
-REM Run cl.exe to find which version our compiler is
-for /f "delims=" %%A in ('cl /? 2^>^&1 ^| findstr /C:"Version"') do set "CL_TEXT=%%A"
-FOR /F "tokens=1,2 delims==" %%i IN ('msvc_versions.bat') DO echo %CL_TEXT% | findstr /C:"Version %%i" > nul && set VSTRING=%%j && goto FOUND
-EXIT 1
-:FOUND
+rem Patches apply only to opencv_contrib
+git apply --whitespace=fix -p0 "%RECIPE_DIR%\kcftracker.patch"
+git apply --whitespace=fix -p0 "%RECIPE_DIR%\ocr_beamsearch_decoder.patch"
+git apply --whitespace=fix -p0 "%RECIPE_DIR%\ocr_hmm_decoder.patch"
+git apply --whitespace=fix -p0 "%RECIPE_DIR%\dpm.patch"
 
-call :TRIM VSTRING %VSTRING%
+mkdir build
+cd build
 
-set OPENCV_ARCH=x86
-if "%VSTRING%" == "9" (
-    set GENERATOR=Visual Studio 9 2008
-    set OPENCV_VC=vc9
-)
-if "%VSTRING%" == "10" (
-    set GENERATOR=Visual Studio 10 2010
-    set OPENCV_VC=vc10
+if "%PY3K%" == "0" (
+    echo "Copying stdint.h for windows"
+    copy "%LIBRARY_INC%\stdint.h" %SRC_DIR%\modules\calib3d\include\stdint.h
+    copy "%LIBRARY_INC%\stdint.h" %SRC_DIR%\modules\videoio\include\stdint.h
+    copy "%LIBRARY_INC%\stdint.h" %SRC_DIR%\modules\highgui\include\stdint.h
```

Comment:
Interesting that it is only a Python 2.x issue. Alright, let's add a comment about this and we can come back to it later.